quipucords/api/management/commands/regenerate_missing_report_cache_files.py

```diff
@@ -0,0 +1,93 @@
+        self.stdout.write(self.style.success(message))
+
+    def find_reports_with_problematic_paths(self) -> set:
+        """Find reports with problematic paths and return their IDs."""
+        report_ids = set()
+        reports_paths = DeploymentsReport.objects.filter(
+            status=DeploymentsReport.STATUS_COMPLETE
+        ).values("id", "cached_fingerprints_file_path", "cached_csv_file_path")
+        self.write_info(
+            f"Found {len(reports_paths)} complete DeploymentsReports to check."
+        )
+        for report_paths in reports_paths:
+            if not report_paths["cached_fingerprints_file_path"]:
+                self.write_warning(
+                    "cached_fingerprints_file_path not set for DeploymentsReport "
+                    f"{report_paths["id"]}"
+                )
+            elif not file_exists(report_paths["cached_fingerprints_file_path"]):
+                self.write_warning(
+                    f"{report_paths["cached_fingerprints_file_path"]} is missing."
+                )
+                report_ids.add(report_paths["id"])
+
+            if not report_paths["cached_csv_file_path"]:
+                self.write_warning(
+                    "cached_csv_file_path not set for DeploymentsReport "
+                    f"{report_paths["id"]}"
+                )
+            elif not file_exists(report_paths["cached_csv_file_path"]):
+                self.write_warning(
+                    f"{report_paths["cached_csv_file_path"]} is missing."
+                )
+                report_ids.add(report_paths["id"])
+
+        return report_ids
+
+    def handle(self, *args, **options):
+        """Handle this command."""
+        report_ids = self.find_reports_with_problematic_paths()
+        failed_report_ids = []
+        if report_ids:
+            report_ids = sorted(list(report_ids))
+            self.write_warning(
+                f"{len(report_ids)} DeploymentReport(s) have missing cached files."
+            )
+            for report_id in report_ids:
+                report = DeploymentsReport.objects.get(pk=report_id)
+                self.write_info(f"Rerunning fingerprint task for {report}")
+                try:
+                    report._rerun_latest_fingerprint(wait=True)
+                except Exception as e:  # noqa: BLE001
+                    self.stdout.write(self.style.ERROR(str(e)))
+                    failed_report_ids.append(report_id)
+        else:
+            self.write_success(
+                "All DeploymentReport cached_fingerprints_file_path and "
+                "cached_csv_file_path files exist."
+            )
+
+        if set(report_ids) - set(failed_report_ids):
+            successes = set(report_ids) - set(failed_report_ids)
+            self.write_success(
+                f"Successfully regnerated files for "
+                f"{len(successes)} DeploymentsReports."
+            )
+        if failed_report_ids:
```

Comment:
If there are failures, please make sure that command returns non-zero status. This will help discovery-ci to decide if we should proceed (and likely encounter issues later), or stop right here.

I guess if we want to be fancy, we can exit 1 when there are any failures, and exit 2 when there are **only** failures. Not sure if there's a point in differentiating these two situations.

Reply:
See also the screenshot in the previous comment. The final red line is the result of invoking Django's `CommandError` which is the prescribed method to automatically cause the command to exit `1`, which I only call if there are nonzero failures. Otherwise, it automatically exits `0`.

I didn't add support for different numbers in different situations, though, as that would require a custom `CommandError` subclass or invoking `sys.exit` directly, but I can add one if you feel strongly about it.

Reply:
Returning non-zero is the one that truly matters, thanks.

I don't feel strongly about differentiating failures. It felt like something that could be useful, so maybe could be done if that's easy enough, but I don't have any immediate use-case in mind.